### PR TITLE
[front] - chore(assistant): disable assistants using disabled providers

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

This PR aims at filtering out assistants using models from providers disabled by the workspace. The changes also apply on templates.

Note that while we want to filter out default/system models we will need other assistants to switch models based on authorized providers. This mechanism will be implemented in a separate PR.

## Risk

This should not have any direct impact on production as all providers are currently enabled.

## Deploy Plan

Deploy `front`

### CODE REVIEW

File: front/components/assistant/AssistantBrowser.tsx [hunk 1]
```diff
@@ -50,10 +51,15 @@ export function AssistantBrowser({
   const router = useRouter();
   const [assistantSearch, setAssistantSearch] = useState<string>("");
 
+  const whiteListedProviders = useMemo(() => {
+    return owner.whiteListedProviders ?? MODEL_PROVIDER_IDS;
```
Comment (spolu) [inline front/components/assistant/AssistantBrowser.tsx:55]: Let's add a function now that we've used this at least in 3 places?

File: front/pages/w/[wId]/assistant/gallery.tsx [hunk 2]
```diff
@@ -83,11 +83,19 @@ export default function AssistantsGallery({
 
   const [assistantSearch, setAssistantSearch] = useState<string>("");
 
+  const filteredAgentConfigurations = useMemo(() => {
```
[the remark this answers is not in the document]
Comment (JulesBelveze) [reply front/pages/w/[wId]/assistant/gallery.tsx:86]: @spolu This is temporary. The intended behaviour is the following: if the admin changes the list of whitelisted providers your assistants will using a deactivated model will switch to another model.

However, I will implement this behaviour in a separate PR as it most likely requires some engineering.

[the remark this answers is not in the document]
Comment (spolu) [reply front/pages/w/[wId]/assistant/gallery.tsx:86]: When will the switch operate ooc? (unrelated to this PR) 

At run time or switch time?

[the remark this answers is not in the document]
Comment (JulesBelveze) [reply front/pages/w/[wId]/assistant/gallery.tsx:86]: Haven't dug enough into it to yet have an opinion. But happy to discuss it once I have a clearer picture 😃

File: front/components/assistant/AssistantBrowser.tsx (hunk 1 above)
Comment (JulesBelveze) [reply front/components/assistant/AssistantBrowser.tsx:55]: @spolu I tried creating the following function:
```typescript
export const getWhiteListedProvidersMemo = (
  workspace: WorkspaceType
) => {
  return useMemo(() => {
    return workspace.whiteListedProviders ?? MODEL_PROVIDER_IDS;
  }, [workspace.whiteListedProviders]);
};
```
But I kept on hitting the following error:
```
Error: FRONT_DATABASE_URI is required but not set
```
as if functions were not called in the right order (rule of hooks from what I gathered?). I change the memoization mechanism in one file, so I'm not sure we need a function anymore?

File: front/pages/w/[wId]/assistant/gallery.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (spolu) [reply front/pages/w/[wId]/assistant/gallery.tsx:86]: I don't think that flies. We can't edit assistants for our users.

I would suggest go the simpler route.

We don't do anything as a first step. And we do have an error that has a very clear message if the assistant is run.

If this comes up often (which I believe won't be the case) we can work on creating some UI to show that the assistant is disabled and should be edited.

**Comment by spolu** [comment]
As per above I would simply create an error condition at run with a very clear message. We can't hide or edit assistants for our users IMHO

**Comment by JulesBelveze** [comment]
Hmmm okay that's what was initially stated in the design doc (being able to switch assistant's model based on white listed providers). But will follow your suggestion and update design doc 👍🏼

File: front/lib/resources/template_resource.ts [hunk 3]
```diff
@@ -132,6 +132,7 @@ export class TemplateResource extends BaseResource<TemplateModel> {
       sId: this.sId,
       tags: this.tags,
       visibility: this.visibility,
+      presetProviderId: this.presetProviderId,
```
Comment (philipperolet) [inline front/lib/resources/template_resource.ts:135]: what is that?

File: front/pages/w/[wId]/builder/assistants/create.tsx [hunk 4]
```diff
@@ -88,15 +89,22 @@ export default function CreateAssistant({
     tags: TemplateTagCodeType[];
   }>({ templates: [], tags: [] });
 
+  const whiteListedProviders = useMemo(() => {
```
Comment (philipperolet) [inline front/pages/w/[wId]/builder/assistants/create.tsx:92]: nit: useMemo isn't necessary here as the function is trivial
(useMemo avoids recomputations at every render and should be used for expensive functions)

File: front/pages/w/[wId]/assistant/gallery.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (philipperolet) [reply front/pages/w/[wId]/assistant/gallery.tsx:86]: Agreed

**Review by philipperolet — COMMENTED** [review]
Thanks! Agreeing with spolu here, we should not edit assistants for users, and let them run with error
Avoiding people thinking "where has my assistant gone" (also left a nit)
Available to discuss when you want if needed :)

**Comment by JulesBelveze** [comment]
New PR: https://github.com/dust-tt/dust/pull/5542

**Pull request closed without merging**